packages/docs/concepts/references-and-templates.yaml

```diff
@@ -24,7 +24,14 @@ _ref:
         type: MarkdownWithCode
         properties:
           content: |
-            References are used to split the configuration of an app into logically distinct files, and to reuse configuration in the app. References can be used almost anywhere in the configuration, as long as the configuration remains valid YAML.
+            References and templates have the following use cases:
+            - Splitting out the config for readability and code navigation.
+            - Splitting out a piece of config in order to use the exact same config in multiple places.
+            - Using a reference with variables to make use of a shared piece of config with dynamic variables.
+            - Using an njk file which allows for much more complicated things - this is an advanced feature and easily overused.
```

Comment:
Lets say something like:
Using a `.njk` file to create config files using the Nunjucks templating language.
Using resolver and transformer functions to create Lowdefy config using Javascript functions.

Reply:
Change made in d9ad6e012a786eeb90ec2e938dd465040c158fb1